apps/test/unit/templates/SignInOrAgeDialogTest.js

```diff
@@ -1,104 +1,85 @@
 
-describe('SignInOrAgeDialog', () => {
-  const defaultProps = {
-    age13Required: true,
-    signedIn: false,
-    storage: new FakeStorage(),
-  };
-
-  before(() => {
-    replaceOnWindow('dashboard', {
-      rack_env: 'unit_test',
-    });
-  });
+const DEFAULT_PROPS = {
+  age13Required: true,
+  signedIn: false,
+  storage: new FakeStorage(),
+};
 
-  after(() => {
-    restoreOnWindow('dashboard');
-  });
+describe('SignInOrAgeDialog', () => {
+  function renderDefault(propOverrides = {}) {
+    render(<SignInOrAgeDialog {...DEFAULT_PROPS} {...propOverrides} />);
+  }
 
   it('renders null if signed in', () => {
-    const wrapper = shallow(
-      <SignInOrAgeDialog {...defaultProps} signedIn={true} />
-    );
-    assert.equal(wrapper.children().length, 0);
+    renderDefault({signedIn: true});
+    expect(screen.queryByText(i18n.signinForProgress())).to.not.exist;
+    expect(screen.queryByText(i18n.provideAge())).to.not.exist;
   });
 
   it('renders null if script does not require 13+', () => {
-    const wrapper = shallow(
-      <SignInOrAgeDialog {...defaultProps} age13Required={false} />
-    );
-    assert.equal(wrapper.children().length, 0);
+    renderDefault({age13Required: false});
+    expect(screen.queryByText(i18n.provideAge())).to.not.exist;
   });
 
   it('renders null if seen before', () => {
-    let getItem = sinon.stub(defaultProps.storage, 'getItem');
+    let getItem = sinon.stub(DEFAULT_PROPS.storage, 'getItem');
     getItem.withArgs('anon_over13').returns('true');
-    const wrapper = shallow(<SignInOrAgeDialog {...defaultProps} />);
-    assert.equal(wrapper.children().length, 0);
+    renderDefault({signedIn: true});
+    expect(screen.queryByText(i18n.signinForProgress())).to.not.exist;
+    expect(screen.queryByText(i18n.provideAge())).to.not.exist;
     getItem.restore();
   });
 
-  it('renders a dialog otherwise', () => {
-    const wrapper = shallow(<SignInOrAgeDialog {...defaultProps} />);
-    assert.equal(wrapper.name(), 'BaseDialog');
-  });
-
   it('renders an explanation and button if under 13', () => {
-    const wrapper = shallow(<SignInOrAgeDialog {...defaultProps} />);
-    const instance = wrapper.instance();
-    instance.ageDropdown = {
-      getValue: () => '12',
-    };
-    assert.equal(wrapper.state().tooYoung, false);
-    instance.onClickAgeOk();
-    assert.equal(wrapper.state().tooYoung, true);
-    assert.equal(
-      wrapper.find('BaseDialog div > div').first().text(),
-      'Tutorial unavailable for younger students'
-    );
-    assert.equal(
-      wrapper.find('BaseDialog Button').props().text,
-      'See all tutorials'
-    );
-    assert.equal(
-      wrapper.find('BaseDialog Button').props().href,
+    renderDefault();
+    fireEvent.change(screen.getByRole('combobox'), {target: {value: '12'}});
+    const okay = screen.getByText('OK');
+    fireEvent.click(okay);
+    expect(screen.queryByText('Tutorial unavailable for younger students')).to
+      .exist;
```

Comment:
Nit: I think this could be simplified to `screen.getByText('Tutorial unavailable for younger students')` without the `expect` since `getBy` will throw an error if it's not found and fail the test

Reply:
Fixed! Thanks!